patterns-use-cases/async-signals-payment/async-signals-payment-java/src/main/java/my/example/PaymentService.java

```diff
@@ -0,0 +1,122 @@
+import my.example.types.PaymentRequest;
+import my.example.utils.PaymentUtils;
+import my.example.utils.StripeUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+//
+// The payment handlers that issues calls to Stripe.
+//  - the result often comes synchronously as a response API call.
+//  - some requests (and some payment methods) only return "processing" and
+//    notify later via a webhook.
+//
+// This example combines both paths in a single function that reliably waits for both
+// paths, if needed, thus giving you a single long-running synchronous function.
+// Durable execution and the persistent awakeable promises combine this into a single
+// reliably promise/async-function.
+//
+// See README on how to run this example (needs a Stripe test account).
+//
+@Service
+public class PaymentService {
+
+  private static final Logger logger = LogManager.getLogger(PaymentService.class);
+  private static final StripeUtils stripe = new StripeUtils();
+
+  @Handler
+  public void processPayment(Context ctx, PaymentRequest request) {
+    PaymentUtils.verifyPaymentRequest(request);
+
+    // Generate a deterministic idempotency key
+    String idempotencyKey = ctx.random().nextUUID().toString();
+
+    // Initiate a listener for external calls for potential webhook callbacks
+    Awakeable<PaymentIntent> webhookPromise = ctx.awakeable(new PaymentIntentSerde());
+
+    // Make a synchronous call to the payment service
+    PaymentIntent paymentIntent =
+        ctx.run(
+            "Stripe call",
+            new PaymentIntentSerde(),
+            () -> {
+              // create payment intent
+              return stripe.createPaymentIntent(
+                  request.getPaymentMethodId(),
+                  request.getAmount(),
+                  idempotencyKey,
+                  webhookPromise.id(),
+                  request.isDelayed());
+            });
+
+    if (!paymentIntent.getStatus().equals("processing")) {
+      // The synchronous call to Stripe had already been completed.
+      // That was fast :)
+      logger.info("Request {} was processed synchronously!", idempotencyKey);
+      PaymentUtils.ensureSuccess(paymentIntent.getStatus());
+    }
+
+    // We did not get the response on the synchronous path, talking to Stripe.
+    // No worries, Stripe will let us know when it is done processing via a webhook.
+    logger.info(
+        "Synchronous response for {} yielded 'processing', awaiting webhook call...",
+        idempotencyKey);
+
+    // We will now wait for the webhook call to complete this promise.
+    // Check out the handler below.
+    PaymentIntent processedPaymentIntent = webhookPromise.await();
+
+    logger.info("Received webhook call for idempotency key: {}", idempotencyKey);
+    PaymentUtils.ensureSuccess(processedPaymentIntent.getStatus());
+  }
+
+  @Handler
+  public boolean processWebhook(Context ctx, @Accept("*/*") @Raw byte[] request) {
```

Comment:
Maybe add a comment on why we use `@Raw` here because we need the stripe SDK to verify the signature.